resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java

```diff
@@ -229,7 +230,20 @@ private Object proceedInThreadPoolBulkhead(ProceedingJoinPoint proceedingJoinPoi
             logger.debug("ThreadPool bulkhead invocation for method {} in backend {}", methodName,
                 backend);
         }
+
         ThreadPoolBulkhead threadPoolBulkhead = threadPoolBulkheadRegistry.bulkhead(backend);
+        if (bulkheadAspectExts != null && !bulkheadAspectExts.isEmpty()) {
+            for (BulkheadAspectExt bulkheadAspectExt : bulkheadAspectExts) {
+                if (bulkheadAspectExt.canHandleReturnType(returnType)) {
+                    return bulkheadAspectExt.handle(
+                            proceedingJoinPoint,
+                            new ThreadPoolBulkheadAdapter(threadPoolBulkhead),
```

Comment:
It was the best approach I was able to come up with, without the need to modify the `BulkheadAspectExt` interface. I thought of casting the `bulkhead` to a `ThreadPoolBulkhead` object but that causes a `ClassCastException` for non-null values.

Reply:
I do see the issues now with the use of the `ThreadPoolBulkheadAdapter`. If I remove the `handleSemaphoreBulkhead` method, it causes the `testAsyncRecovery` method inside of `BulkheadRecoveryTest` to fail. I can create a different implementation for `ThreadPoolBulkheads` rather than using an adapter.